common/class.tx_dlf_solr.php

```diff
@@ -365,23 +370,31 @@ public function search($query = '') {
         while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
 
             $sorting[$resArray['index_name']] = $resArray['index_name'].'_sorting';
-            $this->params['fl'] .= ','.$sorting[$resArray['index_name']];
+            $this->params['fields'] .= ','.$sorting[$resArray['index_name']];
 
         }
 
+        // Set additional query parameters.
+        $this->params['start'] = 0;
+        $this->params['rows'] = $this->limit;
+
+        // Set query.
+        $this->params['query'] = $query;
+
         // Perform search.
-        $results = $this->service->search((string) $query, 0, $this->limit, $this->params);
+        $selectQuery = $this->service->createSelect($this->params);
+        $results = $this->service->select($selectQuery);
 
-        $this->numberOfHits = count($results->response->docs);
+        $this->numberOfHits = min($results->getNumFound(), $this->limit);
```

Comment:
`$this->numberOfHits` should be the number of documents found by the query, not only the number of documents retrieved. If the service did find more documents than it is allowed to retrieve (because of a set limit), we should find another way to communicate this to the users. It would be misleading to tell the users, their query found exactly `limit` documents...

Reply:
Ok, that was a misinterpretation of me.

Reply:
```suggestion
        $this->numberOfHits = $results->getNumFound();
```
Then how about something like this?